rfcs/20221006-frontend_event_tracking.md

```diff
@@ -0,0 +1,469 @@
+   *   - Which media types are the most popular?
+   *   - Do most users search from the homepage, or internal searchbar?
+   */
+  SUBMIT_SEARCH: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+  };
+  /** Description: The user navigates to a page.
+   * Questions:
+   *   - Which content pages are the most popular?
+   *   - Are search result pages commonly shared?
+   *   - If so, which are most popular?
+   *   - What is the most common user path?
+   */
+  VIEW_PAGE: {
+    /** The title of the page being navigated to (window.document.title)  */
+    title: string;
+    /** The path of the previous page */
+    previous: string;
+    /** The path of the new page */
+    next: string;
+  };
+  /**
+   * Description: Whenever the user sets a filter
+   * Questions:
+   *  - Do most users filter their searches?
+   *  - What % of users use filtering?
+   *  - Which filters are most popular? Least popular?
+   *  - Are any filters so commonly applied they should become defaults?
+   */
+  APPLY_FILTER: {
+    /** The name of the filter (not the user-facing element label) */
+    key: string;
+    /** The value of the filter  */
+    value: string | boolean;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+  };
+  /**
+   * Description: Whenever a user changes the content type
+   * Questions:
+   *   - Which content types are most popular?
+   *   - Is there interest in the non-supported content types?
+   *   - Do users switch content types? Where in their journeys?
+   */
+  CHANGE_CONTENT_TYPE: {
+    /** The previously-set media type */
+    previous: MediaType;
+    /** The new media type */
+    next: MediaType;
+    /** The name of the Vue component used to switch content types. */
+    component: string;
+  };
+  /**
+   * Description: Whenever the user scrolls to the end of the results page.
+   * Useful to evaluate how often users load more results or click
+   * on the external sources dropdown.
+   *
+   * This event is mainly used as part of a funnel leading to a
+   * `LOAD_MORE` or `VIEW_EXTERNAL_SOURCES` event.
+   *
+   * Questions:
+   *   - Do users use meta search after reaching the result end?
+   *   - Do users find a result before reaching the end of the results?
+   */
+  REACH_RESULT_END: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The current page of results the user is on. */
+    resultPage: number;
+  };
+  /**
+   * Whenever a search results in less than a full page
+   * of results or no results.
+   *
+   * Questions:
+   *   - How often do searches provide limited or no results?
+   *   - Do users visit external sources when results are insufficient?
+   *   - What do users do after a search has no results?
+   */
+  INSUFFICIENT_RESULTS: {
```

Comment:
All results view is an edge case here: do we consider results 'insufficient' if only one of the media types has less than a full page of results? Maybe we can specifically add which results are insufficient in this case?

Reply:
In this case, I'd only feel it necessary to report if the _aggregated set_ of results was insufficient.